### ISSUE TITLE
MembersIndex Issue - Lost custom fields after umbraco version upgrade

### ISSUE THREAD

**Issue by njvicente82** [umbraco/Umbraco-CMS#13313]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

Version 10.3.1

### Bug summary

After the upgrade of umbraco from version 10.2.1 to 10.3.1 we lost all custom fields in MembersIndex.

Before update:
![image](https://user-images.githubusercontent.com/97965997/198329727-fd814869-64fe-4b2d-bfa7-c04697b12de4.png)

After the update and index rebuild in backoffice:
![image](https://user-images.githubusercontent.com/97965997/198330965-63fa804c-3ed0-449f-b49e-32618845c8c8.png)

### Steps to reproduce

Upgrade of umbraco from version 10.2.1 to 10.3.1 and rebuild MembersIndex

---
_This item has been added to our backlog [AB#24837](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/24837)_

**Comment by Zeegaan** [comment]
Hey @njvicente82, could you make a little code-example where you just add 1 custom field to the member index, so I can reproduce this easily? 🤔 


Also seems like we just have much less fields in 10.3 for some reason, we have 19 fields in 10.2, but only 9 in 10.3, that doesn't seem intentional🤔 
Here is 10.3: 
![image](https://user-images.githubusercontent.com/70372949/205275216-15363994-ff96-43f6-8fdc-17cd34ba3c5b.png)
And 10.2:
![image](https://user-images.githubusercontent.com/70372949/205276148-2464bcd2-95af-4352-8739-53e4ccdd7ca4.png)

**Comment by njvicente82** [comment]
Hey, @Zeegaan 
"Also seems like we just have much less fields in 10.3 for some reason, we have 19 fields in 10.2, but only 9 in 10.3, that doesn't seem intentional" - This is the problem we are reporting

**Comment by nikolajlauridsen** [comment]
Hey @njvicente82 I'm getting a bit confused here, what is the issue you're reporting? is it that your custom fields disappeared, or that default fields disappeared?

**Comment by njvicente82** [comment]
Hey,
In our tests, we added fields to a certain memberType and after the version update these fields stopped appearing in the index as well as some default fields.
Ex: the field firstName:
![image](https://user-images.githubusercontent.com/97965997/205943899-70e94b16-18ef-46b7-a0a2-1a94a5054e88.png)

In version 10.2.1 the field appear in the index
![image](https://user-images.githubusercontent.com/97965997/205944356-2308b080-0a61-4bee-8690-a9149b034ea2.png)
In version 10.3.1 this field is not in the index.

We have a project with 74 fields count and after the update the field count is 11.

I apologize if I am not being clear enough.

**Comment by nikolajlauridsen** [comment]
Ahhh okay, I think I get it now, you've added some properties to the members, which by default got indexed and therefore showed as fields in examine, and after the update the examine fields disappeared.

I'm done for the day, but I'll have a look at reproducing this tomorrow 😀👍

**Comment by nikolajlauridsen** [comment]
I just had a go at reproducing this, and I can indeed produce it 👍 

It seems like for some reason member properties are no longer getting indexed, this is, of course, a regression, and we'll need to have a look at correcting this, so I'll mark it as a sprint candidate, thanks for reporting the issue 😄 🐛 🕵️ 

For when it gets time to fix this I'll add some repro steps 😄 

To reproduce: 

1. Create a fresh V10.3.2 site
2. Create a custom member type, with additional properties 
3. Create a member using that type
4. Go to the examine management dashboard, and find the member 
5. Note how none of the custom properties are present in the index. 

These properties were present in 10.2.1

**Comment by bergmania** [comment]
I have finally figured out what is going on. Seems like you have been relying on a bug in Examine 3.0, which was fixed in Examine 3.0.1 (updated in Umbraco 10.3.0).

It is not intended member properties automatically is added to the member index, due to potential leaking sensitive data.

Since Umbraco 10.3.0 (and before 10.0.0), you have to configure what fields are included in the member index.

This can quite easily be done by replacing the default implementation of `IUmbracoIndexConfig`, with a custom one.

```cs
using Examine;
using Umbraco.Cms.Core;
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.Scoping;
using Umbraco.Cms.Core.Services;
using Umbraco.Cms.Infrastructure.Examine;

namespace Whatever;

public class CustomUmbracoIndexConfigComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
        builder.Services.AddUnique<IUmbracoIndexConfig, MyCustomUmbracoIndexConfig>();
    }
}

public class MyCustomUmbracoIndexConfig : IUmbracoIndexConfig
{
    public MyCustomUmbracoIndexConfig(IPublicAccessService publicAccessService, IScopeProvider scopeProvider)
    {
        ScopeProvider = scopeProvider;
        PublicAccessService = publicAccessService;
    }

    protected IPublicAccessService PublicAccessService { get; }

    protected IScopeProvider ScopeProvider { get; }

    public IContentValueSetValidator GetContentValueSetValidator() =>
        new ContentValueSetValidator(false, true, PublicAccessService, ScopeProvider);

    public IContentValueSetValidator GetPublishedContentValueSetValidator() =>
        new ContentValueSetValidator(true, false, PublicAccessService, ScopeProvider);

    public IValueSetValidator GetMemberValueSetValidator() => new MemberValueSetValidator(
        null,
        null,
        MemberValueSetValidator.DefaultMemberIndexFields.Union(new []
        {
            Constants.Conventions.Member.Comments,
            // Whatever you wanna have in your member index
        } ),
        null);
}
```

In this code sample I add a `CustomUmbracoIndexConfigComposer` that replaces the implementation of `IUmbracoIndexConfig` with my new `MyCustomUmbracoIndexConfig`. This new `MyCustomUmbracoIndexConfig` makes a union on the `MemberValueSetValidator.DefaultMemberIndexFields` with those aliases I wanna have in the index too. In my case only the Comment, with alias: `Constants.Conventions.Member.Comments`.

I'm going to close the issue, as the behaviour is now equal to in Umbraco 9 again.
